tests/functions/folding/test_keccak_sha.py

```diff
@@ -0,0 +1,97 @@
+from hypothesis import (
+    given,
+    settings,
+    strategies as st,
+)
+import pytest
+
+from vyper import (
+    ast as vy_ast,
+    functions as vy_fn,
+)
+
+alphabet = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$%&()*+,-./:;<=>?@[]^_`{|}~'  # NOQA: E501
+
+
+@pytest.mark.fuzzing
+@given(value=st.text(alphabet=alphabet, min_size=0, max_size=100))
+@settings(max_examples=50, deadline=1000)
+@pytest.mark.parametrize('fn_name', ['keccak256', 'sha256'])
+def test_string(get_contract, value, fn_name):
+    source = f"""
+@public
+def foo(a: string[100]) -> bytes32:
+    return {fn_name}(a)
+    """
+    contract = get_contract(source)
+
+    vyper_ast = vy_ast.parse_to_ast(f"{fn_name}('''{value}''')")
+    old_node = vyper_ast.body[0].value
+    new_node = vy_fn.DISPATCH_TABLE[fn_name].evaluate(old_node)
+
+    assert f"0x{contract.foo(value).hex()}" == new_node.value
+
+
+@pytest.mark.fuzzing
+@given(value=st.text(alphabet=alphabet, min_size=0, max_size=100))
```

Comment:
Why not use bytes and then convert to a string?

Reply:
That..  is a great idea.

Reply:
Unfortunately it just led to a bunch of confusion. Some bytes can't decode to a string, and some decoded values such as `\` cause issues.  The exclusion logic ended up being far more confusing than just using `alphabet` and I'm not even sure if I handled all cases.  So sticking with `alphabet`.